hil/api.py

```diff
@@ -682,18 +682,24 @@ def headnode_detach_network(headnode, hnic):
 @rest_call('GET', '/networks', Schema({}))
 def list_networks():
     """Lists all networks"""
-
-    get_auth_backend().require_admin()
-
-    networks = db.session.query(model.Network).all()
     result = {}
-    for n in networks:
-        if n.access:
-            net = {'network_id': n.network_id,
-                   'projects': sorted([p.label for p in n.access])}
-        else:
-            net = {'network_id': n.network_id, 'projects': None}
-        result[n.label] = net
+    # Admin Operation
+    if get_auth_backend().have_admin():
+        networks = db.session.query(model.Network).all()
+        for n in networks:
+            if n.access:
+                projects = sorted([p.label for p in n.access])
+                result[n.label] = {'network_id': n.network_id,
+                                   'projects': projects}
+            else:
+                result[n.label] = {'network_id': n.network_id,
+                                   'projects': None}
```

Comment:
So I'm looking at this again, and we could clean this up a lot doing:

```python
if get_auth_backend.have_admin():
    networks = ...
else:
    networks = ...

# Current admin-only block.
```

Reply:
sure, make sense to me for using the same variable names. I used a different one since originally `networks` is not within the "if" scope.